### ISSUE TITLE
Trouble upgrading to latest version of myNode

### ISSUE THREAD

**Issue by blueredpen** [mynodebtc/mynode#626]
**Describe the bug**
I'm noticing that there's an issue upgrading to the latest version of myNode.

**Expected behavior**

**Screenshots**
++ case ":${PATH}:" in
++ export PATH=/root/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ PATH=/root/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ rustup toolchain list
+ TOOLCHAINS='no installed toolchains'
+ for toolchain in $TOOLCHAINS
+ [[ no == *\l\i\n\u\x* ]]
+ for toolchain in $TOOLCHAINS
+ [[ installed == *\l\i\n\u\x* ]]
+ for toolchain in $TOOLCHAINS
+ [[ toolchains == *\l\i\n\u\x* ]]
+ '[' 1 = 1 ']'
+ '[' 1 = 0 ']'
++ python3 --version
+ CURRENT_PYTHON3_VERSION='Python 3.8.9'
+ [[ Python 3.8.9 != *\P\y\t\h\o\n\ \3\.\8\.\9* ]]
+ echo 'Python up to date'
Python up to date
+ pip3 install --upgrade pip setuptools wheel
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: pip in /usr/local/lib/python3.8/site-packages (22.0)
ERROR: Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 165, in exc_logging_wrapper
    status = run_func(*args)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 339, in run
    requirement_set = resolver.resolve(
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 94, in resolve
    result = self._result = resolver.resolve(
  File "/usr/local/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 481, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/usr/local/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 348, in resolve
    self._add_to_criteria(self.state.criteria, r, parent=None)
  File "/usr/local/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 172, in _add_to_criteria
    if not criterion.candidates:
  File "/usr/local/lib/python3.8/site-packages/pip/_vendor/resolvelib/structs.py", line 151, in __bool__
    return bool(self._sequence)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 155, in __bool__
    return any(self)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 143, in <genexpr>
    return (c for c in iterator if id(c) not in self._incompatible_ids)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 90, in _iter_built_with_inserted
    for version, func in infos:
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 294, in iter_index_candidate_infos
    result = self._finder.find_best_candidate(
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 868, in find_best_candidate
    candidates = self.find_all_candidates(project_name)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 809, in find_all_candidates
    page_candidates = list(page_candidates_it)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/sources.py", line 134, in page_candidates
    yield from self._candidates_from_page(self._link)
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 773, in process_project_url
    page_links = list(parse_links(html_page, self._use_deprecated_html5lib))
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 310, in wrapper_wrapper
    return list(fn(page, use_deprecated_html5lib))
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 350, in parse_links
    parser.feed(page.content.decode(encoding))
  File "/usr/local/lib/python3.8/html/parser.py", line 111, in feed
    self.goahead(0)
  File "/usr/local/lib/python3.8/html/parser.py", line 179, in goahead
    k = self.parse_html_declaration(i)
  File "/usr/local/lib/python3.8/html/parser.py", line 270, in parse_html_declaration
    self.handle_decl(rawdata[i+2:gtpos])
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 405, in handle_decl
    self._raise_error()
  File "/usr/local/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 426, in _raise_error
    raise ValueError(
ValueError: HTML doctype missing or incorrect. Expected <!DOCTYPE html>.

If you believe this error to be incorrect, try passing the command line option --use-deprecated=html5lib and please leave a comment on the pip issue at https://github.com/pypa/pip/issues/10825.

**Desktop (please complete the following information):**

**MyNode hardware (please complete the following information):**
 - Device: [Raspi4]
 - Version [0.2.48]

**Additional context**
Was upgrading manually according to the READme.md

**Comment by tehelsper** [comment]
I just upgraded my 32-bit and 64-bit RP4 systems and didn't see the issue. both did upgrade pip to 22.0.1.

```
admin@myNode:~ $ pip3 --version
pip 22.0.1 from /usr/local/lib/python3.8/site-packages/pip (python 3.8)
```

```
admin@myNode:~ $ pip3 --version
pip 22.0.1 from /usr/local/lib/python3.8/site-packages/pip (python 3.8)
```

If you retry the upgrade, does it fail again?

**Comment by blueredpen** [comment]
Yes, it does. Same error.

**Comment by blueredpen** [comment]
The machine is now in an unusable state. Anything else I should try to do before just wiping + reinstalling?

**Comment by tehelsper** [comment]
You could try updating the line to this:

`pip3 install --no-cache --upgrade pip==21.3.1 setuptools wheel`

**Comment by blueredpen** [comment]
Trying...

**Comment by blueredpen** [comment]
Where can I update the line?

**Comment by blueredpen** [comment]
It's also weird that I'm seeing my version for myNode is set to 0.2.51. But the upgrade hasn't completed...

**Comment by blueredpen** [comment]
Ok, I see it, trying now.

**Comment by blueredpen** [comment]
No, still didn't work.

**Comment by blueredpen** [comment]
Hmmm, so I just restarted the device and I'm now running 2.51. Worked-out ok

**Comment by tehelsper** [comment]
It will report as v0.2.51 as long as the download succeeds. If the upgrade is still failing, you may not be running the versions of software included in v0.2.51.
